stan/math/prim/prob/pareto_type_2_cdf.hpp

```diff
@@ -38,15 +30,20 @@ return_type_t<T_y, T_loc, T_scale, T_shape> pareto_type_2_cdf(
                          alpha);
   check_greater_or_equal(function, "Random variable", y, mu);
 
+  if (size_zero(y, mu, lambda, alpha)) {
+    return 1.0;
+  }
+
+  T_partials_return P(1.0);
```

Comment:
[optional]
This variable should be declared/defined just before the loop where it is used.